elyses-destructured-enchantments/enchantments.js

```diff
@@ -0,0 +1,72 @@
+/// <reference path="./global.d.ts" />
+// @ts-nocheck
+
+/**
+ * Get the first card in the given deck
+ *
+ * @param {Card[]} deck
+ *
+ * @returns {Card} the first card in the deck
+ */
+export function getFirstCard(deck) {
+  const amount = deck;
+  return amount[0];
+}
+
+/**
+ * Get the second card in the given deck
+ *
+ * @param {Card[]} deck
+ *
+ * @returns {Card} the second card in the deck
+ */
+export function getSecondCard(deck) {
+  const amount = deck;
+  return amount[1];
+}
+
+/**
+ * Switch the position of the first two cards in the given deck
+ *
+ * @param {Card[]} deck
+ *
+ * @returns {Card[]} new deck with reordered cards
+ */
+export function swapTopTwoCards(deck) {
+  // eslint-disable-next-line no-param-reassign
+  [deck[0], deck[1]] = [deck[1], deck[0]];
```

Comment:
Please reimplement this solution so that we don't end up mutation the `deck` param. You may use this snippet as a reference:

```
const deck = [8, 4, 3, 5, 1];
const [first, second, ...rest] = deck;
return [second, first, ...rest];
```